src/game/option/option_control.c

```diff
@@ -611,16 +682,18 @@ void Option_Control(INVENTORY_ITEM *inv_item)
             || (g_InputDB.select && m_ControlMenu.cur_role == KC_TITLE)) {
             Option_ControlShutdownText();
             m_KeyMode = KM_INACTIVE;
-            return;
+            g_Input = (INPUT_STATE) { 0 };
+            g_InputDB = (INPUT_STATE) { 0 };
+            return CM_PICK;
         }
 
         if ((g_InputDB.left || g_InputDB.right)
             && m_ControlMenu.cur_role == KC_TITLE) {
-            Option_ControlChangeLayout();
+            layout_num = Option_ControlChangeLayout(mode);
         }
 
         if (g_InputDB.select) {
```

Comment:
One thing I've noticed is the user can press action on the keyboard or controller to enter the remap mode no matter if they are on the keyboard or controller page. They can't actually remap a key with a button and vice versa because of this check:
`if (Input_ReadAndAssignKey(mode, layout_num, m_ControlMenu.cur_role)) {`

but, the game will be stuck waiting for an input on the appropriate device. Not a huge deal but something I should probably fix. I'm not sure how to differentiate where the input game from however since `S_Input_GetCurrentState` checks keyboard and controller input. Any ideas?

Reply:
I don't think this is an issue.